docs/linux/fs.md

```diff
@@ -143,99 +151,103 @@ to the FULL state (finish zone operation).
+can result in I/O errors.
 
 * A zone may transition to the read-only condition:
-  While the data already written in the zone is still readable, the zone can
-  no longer be written. No user action on the zone (zone management command or
-  read/write access) can change the zone condition back to a normal read/write
-  state. While the reasons for the device to transition a zone to read-only
-  state are not defined by the standards, a typical cause for such transition
-  would be a defective write head on an HDD (all zones under this head are
-  changed to read-only).
+  Although the data that is already written in the zone is still readable, the 
+  zone can no longer be written. No user action on the zone (zone management 
+  command or read/write access) can change the zone condition back to a
+  normal read/write state. While the reasons for the device to transition a 
+  zone to read-only state are not defined by the standards, a typical cause 
+  for such transition would be a defective write head on an HDD (all zones 
+  under this head are changed to read-only).
 
 * A zone may transition to the offline condition:
-  An offline zone cannot be read nor written. No user action can transition an
-  offline zone back to an operational good state. Similarly to zone read-only
-  transitions, the reasons for a drive to transition a zone to the offline
-  condition are undefined. A typical cause would be a defective read-write head
-  on an HDD causing all zones on the platter under the broken head to be
-  inaccessible.
+  An offline zone can be neither read nor written. No user action can 
+  transition an offline zone back to an operational "good state". Similar to 
+  zone read-only transitions, the reasons that a drive transitions a zone 
+  to the offline condition are undefined. A typical cause is (for example) a 
+  defective read-write head on an HDD that causes all zones on the platter 
+  under the broken head to be inaccessible.
 
 * Unaligned write errors:
-  These errors result from the host issuing write requests with a start sector
-  that does not correspond to a zone write pointer position when the write
-  request is executed by the device. Even though *zonefs* enforces sequential
-  file write for sequential zones, unaligned write errors may still happen in
-  the case of a partial failure of a very large direct I/O operation split into
-  multiple BIOs/requests or asynchronous I/O operations.  If one of the write
-  request within the set of sequential write requests issued to the device
-  fails, all write requests queued after it will become unaligned and fail.
+  These errors result from the host issuing write requests that have a start 
+  sector that does not correspond to a zone write-pointer position when the 
+  write request is executed by the device (TODO=this sentence is trying to do
+  one thing too many). Although *zonefs* enforces sequential file write for 
```

Comment:
Simplify:

These errors result from the device receiving a write request that has a start sector that does not correspond to the write pointer position of the target zone.

Reply:
Damien, that simplified sentence is a beauty.